ways with more than 2000 nodes: prevent lockup and provide validator-fix

tyrasd opened the pull request:
closes #7381:

1. prevents the locked-up state after the API returns a HTTP 400 status code after uploading a way with too many nodes
2. adds a validator which checks for such objects
3. the validator provides an easy fix by splitting the affected object into small enough pieces:

![image](https://user-images.githubusercontent.com/1927298/141812144-3923f5bd-8a64-4cec-b543-6ddbfcf8b467.png)

Maybe the exact wording could still be improved?

tyrasd committed:
add validation & fix for way vertices limit imposed by OSM API

tyrasd committed:
add new validator to documentation

bhousel commented on modules/validations/osm_api_limits.js:
```diff
@@ -0,0 +1,62 @@
+                var numberOfParts = Math.ceil(entity.nodes.length / maxWayNodes);
+                var splitVertices = [...Array(numberOfParts - 1)].map((_, i) =>
+                    entity.nodes[Math.floor(entity.nodes.length * (i + 1) / numberOfParts)]);
+                var operation = operationSplit(context, splitVertices.concat(entityId));
```
I'd try to split at junctions if you can, rather than where the math happens to divide the way.

tyrasd committed:
add to changelog

tyrasd committed:
fix splitting closed ways at two or more nodes

this bug resulted sometimes in one extra split point in the result, or one of the split vertices to be left unsplit

tyrasd committed:
load maximum way nodes during validation instead of when initializing

that's because the actual (non placeholder) value for the maxWayNode setting will be only populated after the OSM API's capabilities call is finished

tyrasd committed:
try to choose better splitting locations for "simple" cases

tyrasd committed:
add basic test for maxWayNodes validator

tyrasd committed:
add tests for validator's fixes

tyrasd committed:
var -> const/let

tyrasd committed:
tweak wording slightly
